hack/ci/e2e-conformance.sh

```diff
@@ -25,47 +25,143 @@ CLUSTER_NAME=${CLUSTER_NAME:-"test1"}
+  gcloud compute target-tcp-proxies delete --project $GCP_PROJECT $CLUSTER_NAME-apiserver --quiet || true
+  gcloud compute backend-services delete --project $GCP_PROJECT --global $CLUSTER_NAME-apiserver --quiet || true
+  gcloud compute health-checks delete --project $GCP_PROJECT $CLUSTER_NAME-apiserver --quiet || true
+  (gcloud compute instances list --project $GCP_PROJECT | grep $CLUSTER_NAME \
+       | awk '{print "gcloud compute instances delete --project '$GCP_PROJECT' --quiet " $1 " --zone " $2 "\n"}' \
+       | bash) || true
+  (gcloud compute instance-groups list --project $GCP_PROJECT | grep $CLUSTER_NAME \
+       | awk '{print "gcloud compute instance-groups unmanaged delete --project '$GCP_PROJECT' --quiet " $1 " --zone " $2 "\n"}' \
+       | bash) || true
+  (gcloud compute firewall-rules list --project $GCP_PROJECT | grep $CLUSTER_NAME \
+       | awk '{print "gcloud compute firewall-rules delete --project '$GCP_PROJECT' --quiet " $1 "\n"}' \
+       | bash) || true
+
+  # cleanup the networks
+  gcloud compute routers nats delete "${CLUSTER_NAME}-mynat" --router-region="${GCP_REGION}" \
+    --router="${CLUSTER_NAME}-myrouter" --quiet || true
+  gcloud compute routers delete "${CLUSTER_NAME}-myrouter" --project="${GCP_PROJECT}" \
+    --region="${GCP_REGION}" --quiet || true
+
   # remove our tempdir
   # NOTE: this needs to be last, or it will prevent kind delete
   if [[ -n "${TMP_DIR:-}" ]]; then
-    rm -rf "${TMP_DIR}"
+    rm -rf "${TMP_DIR}" || true
   fi
 }
 
+# SSH to a node by name ($1) and run a command ($2).
+function ssh-to-node() {
+  local node="$1"
+  local zone="$2"
+  local cmd="$3"
+
+  # ensure we have an IP to connect to
+  gcloud compute --project "${GCP_PROJECT}" instances add-access-config "${node}" || true
+
+  # Loop until we can successfully ssh into the box
+  for try in {1..5}; do
+    if gcloud compute ssh --ssh-flag="-o LogLevel=quiet" --ssh-flag="-o ConnectTimeout=30" \
+      --project "${GCP_PROJECT}" --zone "${zone}" "${node}" --command "echo test > /dev/null"; then
+      break
+    fi
+    sleep 5
+  done
+  # Then actually try the command.
+  gcloud compute ssh --ssh-flag="-o LogLevel=quiet" --ssh-flag="-o ConnectTimeout=30" \
+    --project "${GCP_PROJECT}" --zone "${zone}" "${node}" --command "${cmd}"
+}
+
 init_image() {
   image=$(gcloud compute images list --project $GCP_PROJECT \
     --no-standard-images --filter="family:capi-ubuntu-1804-k8s" --format="table[no-heading](name)")
   if [[ -z "$image" ]]; then
+      if ! command -v ansible &> /dev/null; then
+        if [[ $EUID -ne 0 ]]; then
+          echo "Please install ansible and try again."
+          exit 1
+        else
+          # we need pip to install ansible
+          curl -L https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+          python get-pip.py --user
+          rm -f get-pip.py
+
+          # install ansible needed by packer
+          version="2.8.5"
+          python -m pip install "ansible==${version}"
```

Comment:
Should this be installed with --user as well?

Reply:
we do this only if we are running as root, so i dropped the --user. see the line above where bailout with ` echo "Please install ansible and try again."`